### PR TITLE
Adding uncontentious methods from RFC FS-1004

### PR DESCRIPTION
**Pull request by wallymathieu**
https://github.com/fsharp/FSharpLangDesign/blob/master/RFCs/FS-1004-result-type.md

```
bind : ('T -> Result<'U, 'TError>) -> Result<'T, 'TError> -> Result<'U, 'TError>
map : ('T -> 'U) -> Result<'T, 'TError> -> Result<'U, 'TError>
mapError : ('TError -> 'U) -> Result<'T, 'TError> -> Result<'T, 'U>
```

The implementation is heavily inspired by how the Option module is structured.

I've added the files in the collections meta folder, since that's where option.fsi and option.fs live (in my mind is somewhat similar).

### CODE REVIEW

**Comment by dsyme** [comment]
This looks good, I'm ok with this addition to the RFC for this release. 

Please add 3-4 unit tests to the FSharp.Core unittests?

**Comment by dsyme** [comment]
AppVeyor lists this failure

```
[00:33:41] 1) Failed : FSharp-Tests-Core+Members+Incremental.incremental(FSI_FILE)
[00:33:41] Error running command 'C:\projects\visualfsharp-3dtit\tests\..\release\net40\bin\fsiAnyCPU.exe' with args ' -r:System.Core.dll --nowarn:20 --define:INTERACTIVE --maxerrors:1 --abortonerror test.fsx' in directory 'C:\projects\visualfsharp-3dtit\tests\fsharp\core\members\incremental'. ERRORLEVEL -1073740771 ERRORLEVEL -1073740771
[00:33:41] at NUnitConf.checkTestResult(Result`2 result) in C:\projects\visualfsharp-3dtit\tests\fsharp\nunitConf.fs:line 14
```

Not sure why, it looks unrelated to your changes.

**Comment by wallymathieu** [comment]
Quite odd indeed.

**Comment by wallymathieu** [comment]
That build looks a bit flaky.

**Comment by wallymathieu** [comment]
Looks like all it needed was some random nudge. ;)

File: src/fsharp/FSharp.Core/result.fsi [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.FSharp.Core
+
+    open Microsoft.FSharp.Core.LanguagePrimitives.IntrinsicOperators
+
+    [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+    module Result =
+
+        /// <summary><c>map f inp</c> evaluates to <c>match inp with Error e -> Error e | Ok x -> Ok (f x)</c>.</summary>
```
Comment (dsyme) [inline src/fsharp/FSharp.Core/result.fsi]: These summaries should be English-language.  I know they're using the style from the "Option" summaries, but those ones should really be changed too :)

The MSDN docs use `summary strings like [`Transforms an option value by using a specified mapping function.``](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/option.map%5B't,'u%5D-function-%5Bfsharp%5D)

**Comment by dsyme** [comment]
Merging, will update the doc strings separately.

**Comment by cartermp** [comment]
:+1:

**Pull request merged**